### ISSUE TITLE
UiKitView has buggy RenderObject

### ISSUE THREAD

**Issue by fzyzcjy** [flutter/flutter#111788]
_Copied from PR_

### How the bug is found

This bug is surely not found by human eyes reading each and every line of Flutter code :)

I proposed the idea that a linter can be created (https://github.com/dart-code-checker/dart-code-metrics/issues/997) to validate whether RenderObject field setters have correct early-return code. @incendial implemented it and ran it (https://github.com/dart-code-checker/dart-code-metrics/pull/1003).

Thus, thanks @incendial for implementing the linter and run it through flutter framework code!

### Bug description

As we know, when implementing a setter in RenderObject, we usually early halt if the new value is equal to the old value (such as [this one](https://github.com/flutter/flutter/blob/7fba37848c9f3007795be2d67a3207067b7a0894/packages/flutter/lib/src/rendering/platform_view.dart#L115), indeed almost all fields in RenderObject child classes are examples). This is very necessary, because we are setting fields in `updateRenderObject` *unconditionally*. If we do not early return, we will execute the full logic like markNeedsPaint and so on *unconditionally* on *every* `updateRenderObject`. That will be performance penalty.

The current PR fixes one bug of such case. In more details, the `viewController` field setter lacks such early-return, and thus unconditionally calls markNeedsPaint. The setter is called from `updateRenderObject` as follows:

![image](https://user-images.githubusercontent.com/5236035/190656619-8aae1809-81f7-475e-8786-73b260c6d3b5.png)

And _UiKitPlatformView is used here:

![image](https://user-images.githubusercontent.com/5236035/190656716-04152bb1-d791-4b0a-8b6a-31f72479ae02.png)

In other words, the controller is not changed in every frame. Instead, in common cases, it should be the same one for many frames. However, it triggers repaint for each and every rebuild.
